### ISSUE TITLE
Install Kodflix on the phone!

### ISSUE THREAD

**Issue by rmallols** [rmallols/kodflix#35]
Our company Director has requested to have Kodflix installed on his phone. We've told him he can access it directly from the mobile web browser, and even have a shortcut on the home screen (with its corresponding icon!) but he definitively wants to go one step further.

So, he wants to receive the app e.g. on the email to get it installed locally.

**Comment by rmallols** [comment]
**TL;DR:**

Getting the app installed locally is relatively easy, thanks to JavaScript. We're going one step further on the rolling process, thanks to the Universal App principle we've been talking so many things.

That said, we shouldn't ignore a couple of things.

1. In reality, we're just installing the `Shell`, also called app container. All the resources (images, data, etc) will remain external. Hence, we're creating a kind of hybrid app where:
* The top layer is native. That has multiple advantages like hardware access (e.g. the camera, fingerprint, etc)
* The UI remains Web. That ensures the app will be always update with the latest content.

2. Even though this process has been notably simplified over years, please note it's still far from being trivial. Because of that, I'd recommend you to keep this as a background task so, if something doesn't work as expected, don't get frustrated and move forward to the next task until you want to revisit it.

**Comment by rmallols** [comment]
**Technical guidance:**

We are going to use [Cordova](https://cordova.apache.org/) as app shell. It's a free, well-known framework with a large community behind.

1. Follow the [get started](https://cordova.apache.org/#getstarted) guide. Please set a meaningful app name (e.g. `kodflix-native`).
* We're installing a 'Hello world' app. We'll see below how to interact with it. 
* Add the platform you want to target (`cordova platform add <platform name>`). 
* If you get any error, try again at the end of the process, once additional dependencies (see below) are installed.

2. Follow the target, platform-specific guide, e.g.
* [Android](https://cordova.apache.org/docs/en/latest/guide/platforms/android/index.html#requirements-and-support) (at least, until `Setting up an Emulator`). 
    * On Mac, please note the `Android SDK` is usually installed on `~/Library/Android/sdk`. 
    * Finding the correct location is necessary to set the `$ANDROID_HOME` variable. 
    * Also note, [while creating the virtual devices](https://developer.android.com/studio/run/managing-avds), to select an API level matching the appropriate Android version (you can get this info by running `cordova platform` from the root folder of the newly created project.
    * You'll have to accept the license agreements at the end (`~/Library/Android/sdk/tools/bin/sdkmanager --licenses`).
    * [Enable USB debugging on your phone](https://www.kingoapp.com/root-tutorials/how-to-enable-usb-debugging-mode-on-android.htm).

* [iOS](https://cordova.apache.org/docs/en/latest/guide/platforms/ios/index.html) (at least, until `Project Configuration`).

3. Prove the 'Hello world' app (created above) works correctly.
* Run `cordova emulate <platform>`, where `platform` is either `android` or `ios`. If everything is OK, it should open a desktop app with an emulated phone running the hello world app.
* Connect your phone to the laptop by cable. Then run `cordova run <platform>`, where `platform` is either `android` or `ios`. If everything is OK, it should open the hello world app on your phone.

4. Let's add some changes to the app, so open it with Visual Studio code.
    * The entry point of the app is `www/index.html` Add an [iframe](https://www.w3schools.com/tags/tag_iframe.asp) pointing to your kodflix app, e.g.
```
        <iframe 
            src='https://kodflix.herokuapp.com' width='100%' scrolling='no' 
            frameBorder='0' id='theiFrame' onLoad='setIframeHeight();' />
```
Remember to change the URL with the location of your kodflix app. Note the `setIframeHeight` bit; that's important to set the height of the frame. Let's add a `script` block on the `head` block:
```
        <script type='text/javascript'>
            function setIframeHeight() {
                let iframe = document.getElementById('theiFrame');
                iframe.style.height = iframe.parentElement.offsetHeight + 'px';
            }
        </script>
```
This will ensure our app will bit all the available space on the screen.
    * Replace the `meta http-equiv="Content-Security-Policy"...` line with this one, to allow rendering external content:
```
<meta http-equiv="Content-Security-Policy" content="default-src * gap: ws: https://ssl.gstatic.com;style-src * 'unsafe-inline' 'self' data: blob:;script-src * 'unsafe-inline' 'unsafe-eval' data: blob:;img-src * data: 'unsafe-inline' 'self' content:;">
``` 
    * On `config.xml`, add the following line to explicitly allow navigation through our kodflix app:
```
<allow-navigation href="https://kodflix.herokuapp.com" />
```
Again, set your Kodflix's URL.
    * Run once again the `emulate` and `run` commands to ensuring everything is OK. You may spot little UI issues with scrolls, paddings, etc. You can fine tune your CSS by changing the `www/css/index.css` file.
